pkg/retry/linear.go

```diff
@@ -0,0 +1,72 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package retry
+
+import (
+	"time"
+)
+
+type linearRetryer struct {
+	retryer
+}
+
+// LinearTicker represents a ticker with a linear algorithm.
+type LinearTicker struct {
+	ticker
+}
+
+// Linear initializes and returns a linear Retryer.
+func Linear(duration time.Duration, setters ...Option) Retryer {
+	opts := NewDefaultOptions(setters...)
+
+	return linearRetryer{
+		retryer: retryer{
+			duration: duration,
+			options:  opts,
+		},
+	}
+}
+
+// NewLinearTicker is a ticker that sends the time on a channel using an
+// linear algorithm.
+func NewLinearTicker(opts *Options) LinearTicker {
+	l := LinearTicker{
+		ticker: ticker{
+			C:       make(chan time.Time, 1),
+			options: opts,
+			stop:    make(chan struct{}, 1),
+		},
+	}
+
+	l.start()
+
+	return l
+}
+
+// Retry implements the Retryer interface.
+func (l linearRetryer) Retry(f RetryableFunc) error {
+	tick := NewLinearTicker(l.options)
+	defer tick.Stop()
+
+	return retry(f, l.duration, tick.C)
+}
+
+func (l LinearTicker) start() {
+	go func() {
+		c := 1.0
+		for {
+			select {
+			case <-l.stop:
+				return
+			default:
+				d := time.Duration(c)*l.options.Units + l.Jitter()
+
+				c++
+				l.C <- time.Now()
+				time.Sleep(d)
```

Comment:
not sure, but sounds like this code won't be stoppable while `Sleep` is running?

It might make sense to refactor this to `<-time.After` in `select` with `<-l.stop`?

Reply:
Addressed in latest changes.